webviews/AddPatternPageApp.vue

```diff
@@ -0,0 +1,227 @@
+  useCommonWebviewState,
+  openFolderExplorer,
+  clearLogs,
+  initializeUI,
+  setupMessageHandler,
+  clearAllFields,
+  createActionWrapper,
+  createFormValidator} from './../src/features/contentCreator/webviewUtils';
+import "../media/contentCreator/addPluginAndPatterPageStyle.css";
+import RequirementsBanner from './RequirementsBanner.vue';
+
+const commonState = useCommonWebviewState();
+const homeDir = commonState.homeDir;
+const logs = commonState.logs;
+
+const initPath = ref("");
+const patternNameTextField = ref("");
+const isOverwritten = ref(false);
+const openScaffoldedFolderButtonDisabled = ref(true);
+const projectUrl = ref("");
+const verboseDropdown = ref("Off");
+
+const canCreate = createFormValidator({
+  patternName: () => patternNameTextField.value.trim() !== ""
+});
+
+const requirementsMet = ref(true);
+const requirementFailures = ref([]);
+
+const handleOpenScaffoldedFolder = () => {
+  vscodeApi.postMessage({
+    type: "init-open-scaffolded-folder-pattern",
+    payload: {
+      projectUrl: projectUrl.value,
+      patternName: patternNameTextField.value.trim(),
+    },
+  });
+};
+
+const handleCreate = createActionWrapper(
+  commonState.isCreating,
+  commonState.logs,
+  commonState.createButtonDisabled,
+  () => {
+    vscodeApi.postMessage({
+      type: "init-add-pattern",
+      payload: {
+        patternName: patternNameTextField.value.trim(),
+        collectionPath: initPath.value.trim() || homeDir.value.trim(),
+        verbosity: verboseDropdown.value.trim(),
+        isOverwritten: isOverwritten.value
+      }
+    });
+  }
+);
+
+const onClear = () => {
+  const componentFields = {
+    patternNameTextField, initPath, verboseDropdown, isOverwritten
+  };
+  const defaults = {
+    verboseDropdown: "Off",
+    isOverwritten: false
+  };
+  clearAllFields(componentFields, defaults);
+  clearAllFields({
+    logs: commonState.logs
+  });
+  commonState.createButtonDisabled.value = false;
+  initializeUI();
+};
+
+onMounted(() => {
+  vscodeApi.postMessage({ type: 'request-requirements-status' });
+  window.addEventListener('message', (event) => {
+    if (event.data && event.data.type === 'requirements-status') {
+      requirementsMet.value = event.data.met;
+      requirementFailures.value = event.data.failures || [];
+    }
+  });
+  setupMessageHandler({
+    onFolderSelected: (data) => {
+      initPath.value = data;
+    },
+    onExecutionLog: (args) => {
+      if (commonState.isCreating.value) {
+        openScaffoldedFolderButtonDisabled.value = args.status !== "passed";
+        projectUrl.value = args.projectUrl || "";
+      }
+    }
+  }, commonState);
+  initializeUI();
+});
+</script>
+
+<template>
+<body>
+  <RequirementsBanner v-if="!requirementsMet" :failures="requirementFailures" />
+  <div :class="{ 'disabled-content': !requirementsMet }">
+  <div class="title-div">
+    <h1>Add a pattern to an existing collection</h1>
+    <p class="subtitle">Extending automation with python</p>
```

Comment:
```suggestion
    <p class="subtitle">Extending automation with python</p>
```
We should change this to something more relevant to patterns. We can ask @tima about this.